executor/aggfuncs/func_max_min.go

```diff
@@ -447,15 +474,17 @@ func (e *maxMin4String) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup
 			// We have to deep copy that string to avoid some potential risks
 			// when the content of that underlying buffer changed.
 			p.val = stringutil.Copy(input)
+			memDelta += int64(len(input))
 			p.isNull = false
 			continue
 		}
 		cmp := types.CompareString(input, p.val, tp.Collate)
 		if e.isMax && cmp == 1 || !e.isMax && cmp == -1 {
 			p.val = stringutil.Copy(input)
+			memDelta += int64(len(input))
```

Comment:
We need to compare the len(p.val) and len(input) to evaluate the memDelta.

Reply:
@mmyj Should I change it to the following?
if cmp ==1 {
    memDelta += int64(len(input))
}

Reply:
I think you should change it like following
oldMem := len(p.val)
newMem := len(input)
memDelta += newMem - oldMem

Reply:
get it

Reply:
note that the json type is the same like this.

Reply:
okay，Whether it needs to redefine a new UpdateMemDeltaGens in aggrunc_test.go

Reply:
I think you should define a new one. The default gens function is always gens 0 memDalta, it is not suitable for string.